SimulationAgent/DependencyResolution.cs

```diff
@@ -72,14 +74,14 @@ private static void SetupCustomRules(ContainerBuilder builder)
             builder.RegisterType<Simulations>().As<ISimulations>().SingleInstance();
             builder.RegisterType<DeviceModels>().As<IDeviceModels>().SingleInstance();
             builder.RegisterType<Services.Devices>().As<IDevices>().SingleInstance();
+            builder.RegisterType<RateLimiting>().As<IRateLimiting>().SingleInstance();
```

Comment:
previously you have a todo for ensuring this is a singleton - is it created anywhere explicitly?  Using DI you're protecting it already it looks like.

Reply:
yes and no, this is kind of tricky. When everything works well, DI will ensure this is a singleton. However, when the singleton container is disposed, this singleton is disposed too which is a bummer because we lose all the counters. Now, that happens only when there are problems while bootstrapping the app, e.g. storage is down, so it's not a big deal right now, but that TODO is to remind me of this issue. I think having a reference in Program.cs will make sure the counters are never reset (something to investigate later).

Reply:
added https://github.com/Azure/device-simulation-dotnet/issues/79

Reply:
>> I think having a reference in Program.cs will make sure the counters are never reset (something to investigate later).

This makes a lot of sense to me - not critical for now though.